domain/network/state/space.go

```diff
@@ -163,17 +145,28 @@ func (st *State) GetSpaceByName(
 	}
 
 	// Append the space.name condition to the query.
-	q := retrieveSpacesStmt + " WHERE space.name = $M.name;"
-
-	s, err := sqlair.Prepare(q, SpaceSubnetRow{}, sqlair.M{})
+	q := `
+SELECT (uuid,
+       name,
+       provider_id,
+       subnet_uuid,
+       subnet_cidr,
+       subnet_vlan_tag,
+       subnet_provider_id,
+       subnet_provider_network_id,
+       subnet_az) AS (&SpaceSubnetRow.*)
+FROM   v_space
+WHERE  name = $M.name;`
+
+	s, err := st.Prepare(q, SpaceSubnetRow{}, sqlair.M{})
 	if err != nil {
 		return nil, errors.Annotatef(err, "preparing %q", q)
 	}
 
 	var rows SpaceSubnetRows
 	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		return errors.Trace(tx.Query(ctx, s, sqlair.M{"name": name}).GetAll(&rows))
```

Comment:
Although we should get away from maps as argument vectors, you might as well instantiate this earlier and supply it to the prepare call.

Reply:
I was on the fence about this map anyway, I've just replaced it with a `Space` and reused that in `Prepare`.